src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/user/controllers/UserController.java

```diff
@@ -328,30 +373,10 @@ public LoginResponse saveUserSettings(@Valid @RequestBody SaveUserSettings saveU
     } catch (SlurFilterReportException | SlurFilterBlockedException e) {
       throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "person_blocked_by_slur_filter");
     }
-    // @todo matrix user
-    person.setShowAvatars(
```

Comment:
This old code would set the value to false if it was previously null. Your new code will only set the value if it was not previously null. If the value was previously null, it will refuse to allow changing it. I don't think that's what you intended?